python/dolfinx/wrappers/fem.cpp

```diff
@@ -761,6 +761,31 @@ void declare_form(nb::module_& m, std::string type)
       .def_prop_ro("integral_types", &dolfinx::fem::Form<T, U>::integral_types)
       .def_prop_ro("needs_facet_permutations",
                    &dolfinx::fem::Form<T, U>::needs_facet_permutations)
+      .def(
+          "set_custom_data",
+          [](dolfinx::fem::Form<T, U>& self, dolfinx::fem::IntegralType type,
+             int id, int kernel_idx, std::optional<std::uintptr_t> data)
+          {
+            self.set_custom_data(type, id, kernel_idx,
+                                 data ? std::optional<void*>((void*)*data)
+                                      : std::nullopt);
+          },
+          nb::arg("type"), nb::arg("id"), nb::arg("kernel_idx"),
+          nb::arg("data").none(),
+          "Set custom data pointer for an integral. The data pointer is "
+          "passed to the kernel. Pass None to clear.")
+      .def(
+          "custom_data",
+          [](const dolfinx::fem::Form<T, U>& self,
+             dolfinx::fem::IntegralType type, int id,
+             int kernel_idx) -> std::optional<std::uintptr_t>
```

Comment:
Why are we returning `std::uintptr_t` and not `void*`? If there is a good reason there needs to be a comment.